lifecycle.go

```diff
@@ -157,6 +158,7 @@ func (d LifeCycle) done() {
 	}
```

Comment:
Yeh, I haven't had a need for this yet. It should be called when tasks of this container are completed, so the done method will deregister itself from consul